### PR TITLE
august: init at unstable-2023-08-13

### PR DESCRIPTION
**Pull request by figsoda**
https://github.com/yoav-lavi/august

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/august/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "yoav-lavi";
+    repo = "august";
+    rev = "42b8a1bf5ca079aca1769d92315f70b193a9cd4a";
+    hash = "sha256-58DZMoRH9PBbM4sok/XbUcwSXBeqUAmFZpffdMKQ+dE=";
+  };
+
+  cargoHash = "sha256-/GvBbsSL0dZ0xTystIpb8sk1nNg5hmP4yceCHlh7EQE=";
+
+  postInstall = ''
+    mv $out/bin/{august-cli,ag}
```
Comment (matthiasbeyer) [inline pkgs/development/compilers/august/default.nix:20]: Why would you do this?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
I am not sure about the rename. `ag` is used in other packages already, for example for the silver searcher (IIRC).

File: pkgs/development/compilers/august/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/development/compilers/august/default.nix:20]: This is what upstream documented: https://github.com/yoav-lavi/august#cli-usage-cli-command-tbd

> ```
> ag -o / --output [OUTPUT] (json|toml|yaml) [INPUT]
> ```

Comment (matthiasbeyer) [reply pkgs/development/compilers/august/default.nix:20]: hm, okay. i suppose its fine then.
I am not sure what the policy on binary name clashes is...

**Review by matthiasbeyer — APPROVED** [review]

**Pull request merged**